src/main/java/mate/academy/controller/BookController.java

```diff
@@ -30,35 +37,86 @@ public class BookController {
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Successfully retrieved book"),
+            @ApiResponse(responseCode = "404", description = "Book not found"),
+            @ApiResponse(responseCode = "500", description = "Internal server error")
+    })
     @GetMapping("/{id}")
-    public BookDto getBookById(@PathVariable Long id) {
+    public BookDto getBookById(@PathVariable @Positive
+                               @Parameter(description = "ID of the book to be retrieved",
+                                       required = true) Long id) {
         return bookService.getBookById(id);
     }
 
+    @Operation(summary = "Delete a book by ID",
+            description = "Deletes the book with the specified ID")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "204", description = "Successfully deleted book"),
+            @ApiResponse(responseCode = "404", description = "Book not found"),
+            @ApiResponse(responseCode = "500", description = "Internal server error")
+    })
     @ResponseStatus(HttpStatus.NO_CONTENT)
     @DeleteMapping("/{id}")
-    public void delete(@PathVariable Long id) {
+    public void delete(@PathVariable @Parameter(description = "ID of the book to be deleted",
+            required = true) Long id) {
         bookService.deleteById(id);
     }
 
+    @Operation(summary = "Update book details by ID",
+            description = "Updates the details of the book with the specified ID")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Successfully updated book"),
+            @ApiResponse(responseCode = "400", description = "Invalid request body"),
+            @ApiResponse(responseCode = "404", description = "Book not found"),
+            @ApiResponse(responseCode = "500", description = "Internal server error")
+    })
     @ResponseStatus(HttpStatus.OK)
     @PutMapping("/{id}")
     public BookDto updateBook(@PathVariable @Positive Long id,
                               @Valid @RequestBody UpdateBookRequestDto requestDto) {
         return bookService.updateBookDetails(id, requestDto);
     }
 
+    @Operation(summary = "Search books by parameters",
+            description = "Searches for books based on provided search parameters")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200",
+                    description = "Successfully retrieved list of books"),
+            @ApiResponse(responseCode = "400",
+                    description = "Invalid search parameters"),
+            @ApiResponse(responseCode = "500",
+                    description = "Internal server error")
+    })
     @GetMapping("/search")
-    public List<BookDto> search(BookSearchParametersDto searchParameters) {
-        return bookService.search(searchParameters);
+    public List<BookDto> search(@Valid BookSearchParametersDto searchParameters,
+                                Pageable pageable) {
```

Comment:
Look at these annotation 
```suggestion
                                @ParameterObject @PageableDefault Pageable pageable) {
```